src/Arcus.WebApi.Security/Authentication/Certificates/BypassCertificateAuthenticationAttribute.cs

```diff
@@ -0,0 +1,11 @@
+﻿using System;
+
+namespace Arcus.WebApi.Security.Authentication.Certificates
+{
+    /// <summary>
+    /// Attribute to circumvent the certificate authentication.
+    /// </summary>
+    public class BypassCertificateAuthenticationAttribute : Attribute 
```

Comment:
Do we actually need an attribute per scenario or do we create a more generic one?

/cc @fgheysels

Reply:
I think it is a good idea to have a specific attribute to bypass the jwt token authentication, as this is used for backdoor security.
I'm wondering if it just isn't enough to specify the `AllowAnonymous` certificate to bypass certificate based authentication ?  (I'm thinking this has more to do with front-door security ? )

Reply:
Yeah, I get that. Maybe we should not include this allow anonymous attribute in our authentication filters.

Reply:
I think we should have a unified approach for all options as it depends on the scenario, no?

I'm not sure if I get what you are saying 🤔

Reply:
Why not? It's a commonly used attribute so I'm not sure why we should not support this? :think

Reply:
Yeah but anonymous is maybe more an authorization term than an authentication term.

Reply:
Here's how .NET Core positions the usage of `AllowAnonymous`:
![image](https://user-images.githubusercontent.com/4345663/96544297-947ca580-12a6-11eb-9a7b-61f357e8284b.png)

https://docs.microsoft.com/en-us/aspnet/core/security/authorization/simple?view=aspnetcore-3.1

This feels similar to what we are adding, no?

Reply:
Yes, and it says "bypasses all authorization statements", it's a thin line of course.

Reply:
So, I would think 'AllowAnonymous' is ok instead of those specific attributes.
(Although I'd like to not use AllowAnonymous to bypass the JwtToken validation.   We might have an endpoint that requires authentication for instance, but doesn't  require backdoor security).
Or am I exaggerating ?

Reply:
Then we add both so you can pick and choose, ok?

Reply:
You mean that applying `AllowAnonymous` will also automatically bypass the backdoor security ?

Reply:
Yes, but if you prefer to use the specific attribute you can still have the JWT token validation for that specific operation.

Those who want to allow anonymous access on, for example a health endpoint, can use AllowAnonymous

Reply:
Ok, I can agree.
Then do we also keep the BypassCertificateAuthenticationAttribute, or is this one completely being replaced by the common AllowAnonymous ?

Reply:
I'd keep those for your scenarios where you only want to bypass certificate auth and not JWT.

But if you say it's no longer required then we should leave it out I think

Reply:
Discussed this with @fgheysels and we've agreed to support `AllowAnonymous` and have a `Bypass{Auth Type}` attribute/filter for more granular control